src/expanduser.cpp

```diff
@@ -35,23 +35,23 @@ MIOPEN_DECLARE_ENV_VAR(HOME)
 
 namespace miopen {
 
-std::string ExpandUser(const std::string& path)
+static std::string GetHomeDir()
 {
-    std::string home_dir;
     const char* const p = GetStringEnv(HOME{});
     if(!(p == nullptr || p == std::string("/") || p == std::string("")))
     {
-        home_dir = std::string(p);
-    }
-    else
-    {
-        // todo:
-        // need to figure out what is the correct thing to do here
-        // in tensoflow unit tests run via bazel, $HOME is not set, so this can happen
-        // setting home_dir to the /tmp for now
-        home_dir = boost::filesystem::temp_directory_path().native();
+        return {p};
     }
+    // todo:
+    // need to figure out what is the correct thing to do here
+    // in tensoflow unit tests run via bazel, $HOME is not set, so this can happen
+    // setting home_dir to the /tmp for now
+    return {boost::filesystem::temp_directory_path().native()};
```

Comment:
`temp_directory_path()` without args may throw exception: An error shall be reported `if !exists(p) || !is_directory(p)`, where p is the path to be returned. This is highly unlikely to happen, but still it's better to use `temp_directory_path(system::error_code& ec)` variant to avoid exceptions.

Reply:
@dmikushin If `temp_directory_path()` fails, then execution environment has really bad issues, similar to cases when malloc fails. The library can't do anything except failing anyway. If you insist, then I can add error handling but that would end with MIOPEN_THROW anyway.